packages/api/src/middleware/validation/checkAnimalOrBatch.js

```diff
@@ -13,81 +13,622 @@
+  creating = true,
+) => {
+  const {
+    default_breed_id,
+    custom_breed_id,
+    breed_name,
+    default_type_id,
+    custom_type_id,
+    type_name,
+  } = animalOrBatch;
+  const breedKeyOptions = ['default_breed_id', 'custom_breed_id', 'breed_name'];
+  const typeKeyOptions = ['default_type_id', 'custom_type_id', 'type_name'];
+
+  // Check if breed is present
+  if (
+    (creating && oneExists(breedKeyOptions, animalOrBatch)) ||
+    oneTruthy([default_breed_id, custom_breed_id, breed_name])
+  ) {
+    checkExactlyOneIsProvided(
+      [default_breed_id, custom_breed_id, breed_name],
+      'default_breed_id, custom_breed_id, or breed_name',
+    );
+  }
+  // Check if breed is present
+  if (!creating && oneExists(breedKeyOptions, animalOrBatch)) {
+    // Overwrite with null in db if editing
+    setFalsyValuesToNull(breedKeyOptions, animalOrBatch);
+  }
+
+  if (
+    oneExists(breedKeyOptions, animalOrBatch) &&
+    !oneTruthy([default_breed_id, custom_breed_id, breed_name])
+  ) {
+    // do nothing if nulling breed
+  } else {
+    // Check if default breed or default type is present
+    if (
+      (oneExists(breedKeyOptions, animalOrBatch) && default_breed_id) ||
+      (oneExists(typeKeyOptions, animalOrBatch) && default_type_id)
+    ) {
+      await checkDefaultBreedMatchesType(animalOrBatchRecord, default_breed_id, default_type_id);
+    }
+    // Check if custom breed or custom type is present
+    if (
+      (oneExists(breedKeyOptions, animalOrBatch) && custom_breed_id && !type_name) ||
+      (oneExists(typeKeyOptions, animalOrBatch) &&
+        (default_type_id || custom_type_id) &&
+        !breed_name)
+    ) {
+      let customBreed;
+      // Find customBreed if exists
+      if (oneExists(breedKeyOptions, animalOrBatch) && custom_breed_id) {
+        checkIdIsNumber(custom_breed_id);
+        customBreed = await CustomAnimalBreedModel.query()
+          .whereNotDeleted()
+          .findById(custom_breed_id);
+        if (!customBreed) {
+          throw customError('Custom breed does not exist');
+        }
+      } else if (animalOrBatchRecord?.custom_breed_id) {
+        checkIdIsNumber(animalOrBatchRecord?.custom_breed_id);
+        customBreed = await CustomAnimalBreedModel.query()
+          .whereNotDeleted()
+          .findById(animalOrBatchRecord.custom_breed_id);
+        if (!customBreed) {
+          // This should not be possible
+          throw customError('Custom breed does not exist');
+        }
+      }
+      // Check custom breed if exists
+      if (customBreed) {
+        await checkRecordBelongsToFarm(customBreed, farm_id, 'custom breed');
+        checkCustomBreedMatchesType(
+          animalOrBatch,
+          animalOrBatchRecord,
+          customBreed,
+          default_type_id,
+          custom_type_id,
+        );
+      }
+    }
+  }
+};
+
+const checkBatchSexDetail = async (
+  animalOrBatch,
+  animalOrBatchKey,
+  animalOrBatchRecord = undefined,
```

Comment:
I asked g pt  since you mentioned "something like"
![Screenshot 2024-10-18 at 9 12 17 PM](https://github.com/user-attachments/assets/171d08a3-1101-499c-a9d3-a79fee0ca35b)

I will change to preexistingAnimalOrBatch

Reply:
Changed!